microraiden/microraiden/client/client.py

```diff
@@ -28,11 +28,10 @@ def __init__(
             channel_manager_address: str = CHANNEL_MANAGER_ADDRESS,
             web3: Web3 = None
     ) -> None:
-        assert private_key or key_path
-        assert not private_key or isinstance(private_key, str)
+        assert private_key is not None or key_path is not None
```

Comment:
I think having just a single variable `private_key` and use a regex to detect hex-string and fallback to `os.path.isfile` if the regex test fails would make it much simpler. Maybe having an absolute path as a requirement would make it even more secure.

Reply:
And another client constructor argument has been eliminated! Nice.